libs/gl-plugin/src/node/mod.rs

```diff
@@ -29,6 +30,32 @@ pub use wrapper::WrappedNodeServer;
 static LIMITER: OnceCell<RateLimiter<NotKeyed, InMemoryState, MonotonicClock>> =
     OnceCell::const_new();
 
+static RPC_CLIENT: OnceCell<Arc<Mutex<cln_rpc::ClnRpc>>> = OnceCell::const_new();
+
+pub async fn get_rpc<P: AsRef<Path>>(path: P) -> Arc<Mutex<cln_rpc::ClnRpc>> {
```

Comment:
Looking good, but I wonder if `OnceCell` is already sufficient to get the semantics we want: the `init` function is guaranteed to be called at most once. This means OnceCell is already enforcing locking behavior. So as long as we can ensure that the first `init` call returns only once the socket exists, that is already sufficient, and we do not need to nest the `Mutex` inside the `OnceCell`.

Let's discuss that during our next meeting, merging this one in the meantime.